apps/root/src/common/components/transaction-steps/index.tsx

```diff
@@ -170,6 +171,23 @@ export type TransactionAction =
   | TransactionActionCreatePosition;
 type TransactionActions = TransactionAction[];
 
+type TransactionActionProps =
+  | TransactionActionApproveTokenProps
+  | TransactionActionApproveTokenSignProps
+  | TransactionActionWaitForApprovalProps
+  | TransactionActionWaitForSimulationProps
+  | TransactionActionWaitForQuotesSimulationProps
+  | TransactionActionSwapProps
+  | TransactionActionCreatePositionProps;
+
+type CommonTransactionActionProps = Pick<
```

Comment:
wouldn't this just be ItemProps?

Reply:
Properties such as `explanation` and `done` props are related to TransactionActionProps and not present in ItemProps. By using TransactionActionProps (extended from ItemProps) I can pick those two and also  `isLast` and `isCurrentStep` from ItemProps.
We could redefine the scope of both interfaces but I found this to be the easiest way to proceed

Reply:
But ItemProps is only used to extend each actionProp so shouldn't done and explanation be there?

Reply:
Oh I think I get it now!